DQM/Physics/src/TopSingleLeptonDQM.cc

```diff
@@ -415,38 +433,53 @@ void MonitorEnsemble::fill(const edm::Event& event,
     reco::GsfElectronRef gsf_el = elec->gsfElectronRef();
     // restrict to electrons with good electronId
     if (electronId_.isUninitialized() ? true : ((double)(*electronId)[gsf_el] >=
-                                                eidCutValue_)) {
+                                                eidCutValue_)) { //This Electron Id is not currently used, but we can keep this for future needs
       if (!elecSelect_ || (*elecSelect_)(*elec)) {
 
         double el_ChHadIso = gsf_el->pfIsolationVariables().sumChargedHadronPt;
         double el_NeHadIso = gsf_el->pfIsolationVariables().sumNeutralHadronEt;
         double el_PhIso = gsf_el->pfIsolationVariables().sumPhotonEt;
-        double el_pfRelIso =
-            (el_ChHadIso +
-             max(0., el_NeHadIso + el_PhIso -
-                         0.5 * gsf_el->pfIsolationVariables().sumPUPt)) /
-            gsf_el->pt();
-        if (eMult == 0) {
-          // restrict to the leading electron
-          fill("elecPt_", elec->pt());
-          fill("elecEta_", elec->eta());
+				double absEta = abs(gsf_el->superCluster()->eta());
+				double eA = 0;				
+				if      (absEta < 1.000) eA = 0.1703;
+				else if (absEta < 1.479) eA = 0.1715;
+				else if (absEta < 2.000) eA = 0.1213;
+				else if (absEta < 2.200) eA = 0.1230;
+				else if (absEta < 2.300) eA = 0.1635;
+				else if (absEta < 2.400) eA = 0.1937;
+				else if (absEta < 5.000) eA = 0.2393;
+	
+				double rho = _rhoHandle.isValid() ? (float)(*_rhoHandle) : 0;
+        double el_pfRelIso = (el_ChHadIso + max(0., el_NeHadIso + el_PhIso - rho * eA)) /gsf_el->pt();
+
+				//Only tightId
+
+				//Tight Iso
+
+
+        if (eMult == 0) {//Leading electron without Iso requirements (tight/loose defined on cfi file)
+          // restrict to the leading tight isolated electron
           fill("elecRelIso_", el_pfRelIso);
           fill("elecChHadIso_", el_ChHadIso);
           fill("elecNeHadIso_", el_NeHadIso);
           fill("elecPhIso_", el_PhIso);
         }
-        // in addition to the multiplicity counter buffer the iso
-        // electron candidates for later overlap check with jets
         ++eMult;
-        if (!elecIso_ || (*elecIso_)(*elec)) {
-          isoElecs.push_back(&(*elec));
-          ++eMultIso;
-        }
+
+				if (!((el_pfRelIso<0.0588 && absEta<1.479)||(el_pfRelIso<0.0571 && absEta>1.479))) continue;
```

Comment:
please add a comment on how these magic numbers were derived

Reply:
add reference below
// PF Isolation with Effective Area Corrections according to https://twiki.cern.ch/twiki/bin/viewauth/CMS/CutBasedElectronIdentificationRun2